src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -124,7 +124,7 @@ float64_t CGaussianARDKernel::get_parameter_gradient_helper(
 
 	if (!strcmp(param->m_name, "weights"))
 	{
-		linalg::add(avec, bvec, bvec, 1.0, -1.0);
+		bvec=linalg::add(avec, bvec, 1.0, -1.0);
```

Comment:
@yorkerlin ah due the usage of this, we're missing the _in-place_ benefits here :(
Shall I add a param to specify whether you want this _in-place_ or not?